dns.go

```diff
@@ -0,0 +1,78 @@
+package warc
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/miekg/dns"
+)
+
+type cachedIP struct {
+	expiresAt time.Time
+	ip        net.IP
+}
+
+func (d *customDialer) archiveDNS(address string) (resolvedIP net.IP, err error) {
+	// Get the address without the port if there is one
+	address, _, err = net.SplitHostPort(address)
+	if err != nil {
+		return resolvedIP, err
+	}
+
+	// Check if the address is already an IP
+	resolvedIP = net.ParseIP(address)
+	if resolvedIP != nil {
+		return resolvedIP, nil
+	}
+
+	// Check cache first
+	if cached, ok := d.DNSRecords.Load(address); ok {
+		cachedEntry := cached.(cachedIP)
+		if time.Now().Before(cachedEntry.expiresAt) {
+			return resolvedIP, nil
+		}
+		// Cache entry expired, remove it
+		d.DNSRecords.Delete(address)
+	}
+
+	m := new(dns.Msg)
+	m.SetQuestion(dns.Fqdn(address), dns.TypeA)
```

Comment:
`dns.TypeAAAA` should be asked too

Reply:
but not in the same request, see : https://github.com/miekg/dns/issues/1092

Reply:
Done! Thank you